applications/GeoMechanicsApplication/custom_elements/triangle_2d_15.hpp

```diff
@@ -0,0 +1,1753 @@
+#pragma once
+//    |  /           |
+//    ' /   __| _` | __|  _ \   __|
+//    . \  |   (   | |   (   |\__ `
+//   _|\_\_|  \__,_|\__|\___/ ____/
+//                   Multi-Physics
+//
+//  License:		 BSD License
+//					 Kratos default license: kratos/license.txt
+//
+//  Main authors:    Mohamed Nabi
+//                   
+//                   
+//  contributors:    
+//                   
+//
+
+
+#if !defined(KRATOS_TRIANGLE_2D_15_H_INCLUDED )
+#define  KRATOS_TRIANGLE_2D_15_H_INCLUDED
+
+// System includes
+
+// External includes
+
+// Project includes
+#include "custom_elements/line_2d_5.hpp"
+#include "integration/triangle_gauss_legendre_integration_points.h"
+
+namespace Kratos
+{
+    ///@name Kratos Globals
+    ///@{
+
+    ///@}
+    ///@name Type Definitions
+    ///@{
+
+    ///@}
+    ///@name  Enum's
+    ///@{
+
+    ///@}
+    ///@name  Functions
+    ///@{
+
+    ///@}
+    ///@name Kratos Classes
+    ///@{
+
+    /**
+     * @class Triangle2D15
+     * @ingroup KratosCore
+     * @brief A fifteen node 2D triangular geometry with quadratic shape functions
+     * @details While the shape functions are only defined in 2D it is possible to define an arbitrary orientation in space. Thus it can be used for defining surfaces on 3D elements.
+     * The node ordering corresponds with:
+     *          2
+     *          |`\
+     *          9  8
+     *          |   `\
+     *          10 14 7
+     *          |      '\
+     *          11 12 13 6
```

Comment:
Great!! What was the problem a few weeks ago that prevented this element to work? :)

Reply:
Thank you @AlejandroCornejo ,
In the Geo core, the number of Gauss points is hard programmed to use Gauss_2 (3 points) for specific type of elements. The 15 nodes element was not converging because of this limitation.